### ISSUE TITLE
Running power flow on large scale TSO system

### ISSUE THREAD

**Issue by Hofsmo** [NREL-Sienna/PowerFlows.jl#14]
Hi, I was wondering if you have experience with running power flow on power systems with some thousands buses and branches? I am working with a TSO and they seemed quite interested in having their system running on an open source software.

As far as I can tell the system does not contain any component types not supported by the library. However, even though I export the raw data with the solution obtained by PSS/E it does not converge. Have you had any similar experiences and have any tips on what I should look for?

Another thing I noticed is that several of the buses have a angles outside the +- 90 degrees, so I need to run the import `runchecks=False` to be able to import the system.

I guess this is more of a question than an issue, but I hope it is a question that is relevant to more than me and that you have some tips. 

Best regards Sigurd

**Comment by jd-lara** [comment]
Hi @Hofsmo we tested PowerFlows in systems with up to 2000 buses from the Texas A&M library. The underlying implementation works fine for well initialized systems but we have noticed some challenges in the underlying NLSolve library with initializations. Also note that the current implementation doesn't include reactive power limits so if the PSSe case has a solution accounting for Q limits is possible that the solution will diverge.  

We have plans to improve the PowerFlowData implementation and capabilities of this library but it isn't unfortunately at the top of the list of efforts we are pursuing in the next couple of months. 

About the buses checks, that's something we can address in PowerSystems.jl when parsing the data.

**Comment by Hofsmo** [comment]
Thanks for the quick reply :). I tried without VAR-limits and also by locking taps and shunts. PSS/E finds a solution, but still no luck with the library. 
I fully understand if the power flow solver is on top of your priority list. I will try to simplify the network a bit and see if I have more success then.

**Comment by Hofsmo** [comment]
Hi,
I finally had time to look at this again and  I managed to make the code run on a large TSO-system. In case anyone else is trying to achieve the same I will give a brief overview here on what I did.
I made a Python code that first reduce the size of the network a bit by removing some of the areas. Then it removes all phase shifting transformers. I do this by reducing the phase shift in steps and running a load flow afterwards untill the phase shift reaches zero. I adjust some of the reactive power limits for some generators too. Afterwards, I sometimes end up with a non convergent case that I fix manually in PSS/E. PSS/E may then change the bus code for some generators to minus -2. These can normally be inserted back in tot the case one by one, although some work may be needed to fix the reactive limits or add compensation.

Could be better ways of doing this, but at least it works :).
